source/common/stats/stats_impl.h

```diff
@@ -447,17 +489,20 @@ class IsolatedStoreImpl : public Store {
       : counters_([this](const std::string& name) -> CounterSharedPtr {
           std::string tag_extracted_name = name;
           std::vector<Tag> tags;
-          return alloc_.makeCounter(name, std::move(tag_extracted_name), std::move(tags));
+          return alloc_->makeCounter(name, std::move(tag_extracted_name), std::move(tags));
         }),
         gauges_([this](const std::string& name) -> GaugeSharedPtr {
           std::string tag_extracted_name = name;
           std::vector<Tag> tags;
-          return alloc_.makeGauge(name, std::move(tag_extracted_name), std::move(tags));
+          return alloc_->makeGauge(name, std::move(tag_extracted_name), std::move(tags));
         }),
         histograms_([this](const std::string& name) -> HistogramSharedPtr {
           return std::make_shared<HistogramImpl>(name, *this, std::string(name),
                                                  std::vector<Tag>());
-        }) {}
+        }) {
+    stats_options_.max_obj_name_length_ = 1000 * 1000;
```

Comment:
Actually I went down the road of having a special sentinal value or out-of-band bool to determine whether it's limitless, but it involves way more changes to the code, with associated risk.

I think 1M in terms of a stat name length is infinite enough, but I could use numeric_limits instead I suppose. I'll look at that and then write "done".

Reply:
I am always leery of going to the absolute max, and that was indeed a concern this time.  I had to divde numeric_limits by 2 to avoid an overflow in the calculation of maxNameLength().

Anyway, done.